### PR TITLE
[Java] classes - new exercise

### PR DESCRIPTION
**Pull request by mikedamay**
All files complete except java/config.json and exercises/settings.gradle which I will update before merging

### CODE REVIEW

**Comment by mirkoperillo** [comment]
Hi @mikedamay sorry for the late, I was offline for my summer holidays
I'll provide you a review ASAP

Thank you for contributing to Java track

File: languages/java/exercises/concept/classes/.docs/after.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+- [`public`][public]: the member can be accessed by any code (no restrictions).
+- [`private`][private]: the member can only be accessed by code in the same class.
+
+It is customary to specify an access modifier for all members. If no access modifier is specified, it will default to `private`.
+
+The above-mentioned grouping of related data and behavior plus restricting access to members is known as [_encapsulation_][encapsulation], which is one of the core object-oriented concepts.
+
+You can think of a class as a template for creating instances of that class. To [create an instance of a class][creating-objects] (also known as an _object_), the [`new` keyword][new] is used:
+
+```csharp
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/.docs/after.md:14]: replace `csharp` with `java`. 
This is valid for other blocks of the page

File: languages/java/exercises/concept/classes/.docs/after.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+The primary object-oriented construct in Java is the _class_, which is a combination of data ([_fields_][fields]), also known as instance variables, and behavior ([_methods_][methods]). The fields and methods of a class are known as its _members_.
+
+Access to members can be controlled through access modifiers, the two most common ones being:
+
+- [`public`][public]: the member can be accessed by any code (no restrictions).
+- [`private`][private]: the member can only be accessed by code in the same class.
+
+It is customary to specify an access modifier for all members. If no access modifier is specified, it will default to `private`.
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/.docs/after.md:8]: In Java if no access modifier is specified, the default one is `package visibility`.
So the member is visible to all classes defined into the same package

File: languages/java/exercises/concept/classes/.docs/after.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+The above-mentioned grouping of related data and behavior plus restricting access to members is known as [_encapsulation_][encapsulation], which is one of the core object-oriented concepts.
+
+You can think of a class as a template for creating instances of that class. To [create an instance of a class][creating-objects] (also known as an _object_), the [`new` keyword][new] is used:
+
+```csharp
+class Car
+{
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/.docs/after.md:16]: normally in Java left curly bracket is on the same line of the class name
so 
```
class Car {
}
```
This is valid for all the code in this page
This is valid for block of code too

File: languages/java/exercises/concept/classes/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+If the battery is at 0%, you can't drive the car anymore and the battery display will show `"Battery empty"`.
+
+You have six tasks, each of which will work with remote controlled car instances.
+
+## 1. Buy a brand-new remote controlled car
+
+Implement the (_static_) `RemoteControlCar.buy()` method to return a brand-new remote controlled car instance:
+
+```csharp
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/.docs/instructions.md:18]: replace `csharp` with `java`. 
This is valid for other blocks of the page

File: languages/java/exercises/concept/classes/.docs/introduction.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
+The primary object-oriented construct in Java is the _class_, which is a combination of data (_fields_) and behavior (_methods_). The fields and methods of a class are known as its _members_.
+
+Access to members can be controlled through access modifiers, the two most common ones being:
+
+- `public`: the member can be accessed by any code (no restrictions).
+- `private`: the member can only be accessed by code in the same class.
+
+You can think of a class as a template for creating instances of that class. To create an instance of a class (also known as an _object_), the `new` keyword is used:
+
+```csharp
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/.docs/introduction.md:10]: replace `csharp` with `java`. 
This is valid for other blocks of the page

File: languages/java/exercises/concept/classes/.docs/introduction.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+class Car
+{
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/.docs/introduction.md:12]: normally in Java left curly bracket is on the same line of the class name
so 
```
class Car {
}
```
This is valid for all the code in this page
This is valid for block of code too

File: languages/java/exercises/concept/classes/.meta/src/reference/java/RemoteControlCar.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
+class RemoteControlCar
+{
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/.meta/src/reference/java/RemoteControlCar.java:2]: normally in Java left curly bracket is on the same line of the class name
so 
```
class Car {
}
```
This is valid for all the code in this page
This is valid for block of code too

File: languages/java/exercises/concept/classes/src/test/java/RemoteControlCarTest.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class RemoteControlCarTest
+{
```
Comment (mirkoperillo) [inline languages/java/exercises/concept/classes/src/test/java/RemoteControlCarTest.java:6]: normally in Java left curly bracket is on the same line of the class name
so 
```
class Car {
}
```
This is valid for all the code in this page
This is valid for block of code too

**Review by mirkoperillo — CHANGES_REQUESTED** [review]
@mikedamay thank you for your help.
The work is good, it needs only some minor changes before the merging

**Comment by mikedamay** [comment]
Thanks for the review - all points have been addressed.  I will update config.json before merging

**Comment by mirkoperillo** [comment]
@mikedamay thank you for dedicating time to this: the markdown part looks good, you miss only fixing the code part and the review can be consider ok

**Comment by mikedamay** [comment]
Done

**Review by mirkoperillo — APPROVED** [review]

**Comment by mirkoperillo** [comment]
@mikedamay thank you, LGTM.

You can add missing stuff in `exercises/settings.gradle` and `java/config.json` and merge it when you want.

**Pull request merged**